ext/datasource/etcdv3/etcdv3.go

```diff
@@ -0,0 +1,114 @@
+package etcdv3
+
+import (
+	"context"
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/mvcc/mvccpb"
+	"github.com/pkg/errors"
+	"time"
+)
+
+var logger = logging.GetDefaultLogger()
+
+type etcdv3DataSource struct{
+	handler []datasource.PropertyHandler
+	client *clientv3.Client
+	ruleKey string
+}
+
+func (c *etcdv3DataSource)AddPropertyHandler(h datasource.PropertyHandler){
+	c.handler = append(c.handler, h)
+}
+
+func (c *etcdv3DataSource)RemovePropertyHandler(h datasource.PropertyHandler){
+	for index, value := range c.handler{
+		if value == h{
+			c.handler = append(c.handler[:index], c.handler[index+1:]...)
+			break
+		}
+	}
+}
+
+func (c *etcdv3DataSource)ReadSource() ([]byte, error){
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	resp, err := c.client.Get(ctx, c.ruleKey)
+	if err != nil{
+		return nil, err
+	}
+	if resp.Count == 0{
+		return nil, errors.Errorf("No %v key in etcd", c.ruleKey)
+	}
+	return resp.Kvs[0].Value, nil
+}
+
+func (c *etcdv3DataSource)Initialize() error{
+	go util.RunWithRecover(c.watch, logger)
+	newValue, err := c.ReadSource()
```

Comment:
Maybe encapsulating the ReadSource and updateValue into one function is more elegant.